### PR TITLE
[1LP][RFR] Ownership: Remove dependency on existing vm

### PR DESCRIPTION
**Pull request by apagac**
This is a refactor of existing ownership tests. The point of this PR is to remove the dependency on a vm set in the cfme_data.yaml.

Before, the ownership tests required a pre-provisioned VM with name set in the cfme_data file in every provider we wanted to test, like so:
"ownership_vm: <name_of_the_vm>"
This was cumbersome, the data repeated across the cfme_data.yaml file and it caused problems when trying to run PRT or automation, because there was no entry "ownership_vm" in the cfme_data.yaml.

Now, the module works with a yield fixture which provisions the VM beforehand, as seen in many other tests that are currently running in automation.

{{ pytest: -v --long-running --use-provider vsphere65-nested cfme/tests/cloud_infra_common/test_vm_ownership.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 02fcd72e1241afffb6c10d589639ec70148463fe:

`cfme/tests/cloud_infra_common/test_vm_ownership.py`:
- :red_circle: Line 31:9: **F821** *undefined name 'logger'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by apagac** [comment]
5.9 failure occurs due to a known bug 1533391

File: cfme/tests/cloud_infra_common/test_vm_ownership.py [hunk 1]
```diff
@@ -8,16 +8,30 @@
...
 
+@pytest.yield_fixture(scope="module")
+def vm_crud(provider):
+    vm = VM.factory(random_vm_name(context='ownrs'), provider)
+    vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+    yield vm
+
+    try:
+        vm.delete_from_provider()
+    except:
```
Comment (rlbabyuk) [inline cfme/tests/cloud_infra_common/test_vm_ownership.py]: **Required**: Specify exception type, at least Exception should be used here

File: cfme/tests/cloud_infra_common/test_vm_ownership.py [hunk 2]
```diff
@@ -8,16 +8,30 @@
...
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
 
 
 pytestmark = [
     test_requirements.ownership,
     pytest.mark.meta(blockers=[BZ(1380781, forced_streams=["5.7"])]),
     pytest.mark.tier(3),
-    pytest.mark.provider([BaseProvider], required_fields=['ownership_vm'], scope='module')
+    pytest.mark.provider([BaseProvider], scope='module')
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_vm_ownership.py]: **Required**

Restrict this to CloudInfraProvider, I don't think you'll want to generate against network providers, physical infra providers, or container providers - yet.

**Pull request merged**